internal/pkg/fleetapi/client/client_test.go

```diff
@@ -183,6 +183,37 @@ func TestElasticApiVersion(t *testing.T) {
 		}
 	})
 
+	t.Run("verify that we don't log a generic 400 status as a downgrade request", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		mux := http.NewServeMux()
+		mux.HandleFunc("/someotheroperationreturnsbadrequest", func(writer http.ResponseWriter, request *http.Request) {
+			assert.Equal(t, request.Header.Get(elasticApiVersionHeaderKey), defaultFleetApiVersion)
```

Comment:
[Blocker]
What exactly do you intend to test here?
Because if for some reason the code was changed and the agent client ins't sending `defaultFleetApiVersion` as the version, it'll be caught at assertion at the end of the test. I'm not sure you need this assert here. I don't believe this test is testing the client indeed sends `defaultFleetApiVersion` as the api version.

Reply:
The agent client now ALWAYS send the defaultFleetApiVersion (it's the only version we support).
This assertion is to verify that we get the correct header value in the request on the server side.

The assertion at the end of the test is to check the behaviour of the client to the header in the response.